frontstage/views/passwords/reset_password.py

```diff
@@ -118,12 +118,12 @@ def request_password_change(email):
 
     party_id = str(respondent["id"])
     password_reset_counter = party_controller.get_password_reset_counter(party_id)["counter"]
-
-    # When the password verification token has expired, it is deleted from the DB. The password counter would always be
-    # >0 and the the try below would be entered (when only the passworrd_reset_counter was in the if). When this
-    # occurred, the code would error as the password_verfication_token would be missing from the dictionary. I've added
-    # a check to ensure that this never enters the try when this is missing from the dict.
-    if password_reset_counter != 0 and "password_verification_token" in respondent:
+    # When the password_verification_token has expired, it will be deleted from the DB, however, the
+    # password_reset_counter is not reset and will be > 0. This led to an error 500 when the below 'if' statement
+    # was entered as a result of the password_verification_token being null. The statement has been updated to check
+    # that there is an actual string present and not null or an empty string. This should minimise the chances of an
+    # error 500 occurring.
+    if password_reset_counter > 0 and respondent.get("password_verification_token"):
```

Comment:
So the code inside the _if_ is basically saying use the email inside the password_verification_token if there is one. We intrinsically know if there is a password_verification_token that the counter must be greater than 1. So with this in mind we should be able to just do this 
```
 if verification_token:= respondent.get("password_verification_token"):
        try:
            email = verification.decode_email_token(verification_token, app.config["PASSWORD_RESET_ATTEMPTS_TIMEOUT"])```

This then doesn't need any comment or just  # When the password verification token has expired, it is deleted from the DB

Reply:
That's a good idea. I had thought about moving this up. It makes sense. I hadn't moved it thinking that if there had been an issue with the reset password that it would mess up the whole reset. But in hindsight, the counter should reset when the timeout regardless and so it shouldn't cause any issues.